prow/flagutil/core_config.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+
+	"k8s.io/test-infra/prow/config"
+)
+
+// CoreConfigOptions holds options for interacting with Configs.
+type CoreConfigOptions struct {
+	ConfigPath    string
+	JobConfigPath string
+}
+
+// AddFlags injects config options into the given FlagSet.
+func (o *CoreConfigOptions) AddFlags(fs *flag.FlagSet) {
+	fs.StringVar(&o.ConfigPath, "config-path", "/etc/config/config.yaml", "Path to config.yaml.")
+	fs.StringVar(&o.JobConfigPath, "job-config-path", "", "Path to prow job configs.")
```

Comment:
How will this support `plank` or other components that want the `config-path` but not the `job-config-path`?

Reply:
This change leaves validation to the individual binaries, so plank would just not validate anything for `job-config-path`. Then the default would be set `""`.

Reply:
But we don't want to expose flags that are not supposed to be used or set, though, right? That was one of the mistakes with the prior DRY flag approaches -- end up with `-deck-url` flag on all sorts of binaries that could never use it

Reply:
Okie dokie, splitting it out.